test/fuzz/kubestellar_fuzz_test.go

```diff
@@ -0,0 +1,403 @@
+			}
+		}
+
+		// Test API group allowance
+		testGroups := []string{"apps", "networking.k8s.io", "control.kubestellar.io", "apiextensions.k8s.io"}
+		for _, group := range testGroups {
+			allowed := util.IsAPIGroupAllowed(group, apiGroups)
+			// If apiGroups is nil or empty, all groups should be allowed
+			if apiGroups == nil || len(apiGroups) == 0 {
+				if !allowed {
+					t.Errorf("Group %q should be allowed when no restrictions are set", group)
+				}
+			}
+		}
+	})
+}
+
+// FuzzJSONValueValidation tests JSON value validation
+func FuzzJSONValueValidation(f *testing.F) {
+	// Add seed corpus for JSON value validation
+	seedCorpus := []string{
+		`{"type": "String", "string": "test"}`,
+		`{"type": "Number", "float": "42"}`,
+		`{"type": "Bool", "bool": true}`,
+		`{"type": "Object", "object": {"key": "value"}}`,
+		`{"type": "Array", "array": [1, 2, 3]}`,
+		`{"type": "Null"}`,
+	}
+
+	for _, seed := range seedCorpus {
+		f.Add(seed)
+	}
+
+	f.Fuzz(func(t *testing.T, jsonStr string) {
+		// Skip empty strings
+		if strings.TrimSpace(jsonStr) == "" {
+			return
+		}
+
+		// Try to parse as JSON
+		var value v1alpha1.Value
+		err := json.Unmarshal([]byte(jsonStr), &value)
+		if err != nil {
+			// Expected for invalid JSON, just return
+			return
+		}
+
+		// Skip validation if the type field is empty or invalid
+		// This happens when JSON has typos in field names (e.g., "tYpe" instead of "type")
+		// or when the type value is not one of the valid ValueType constants
+		if value.Type == "" {
+			return
+		}
+
+		// Validate that the type is one of the known valid types
+		validTypes := map[v1alpha1.ValueType]bool{
+			v1alpha1.TypeString: true,
+			v1alpha1.TypeNumber: true,
+			v1alpha1.TypeBool:   true,
+			v1alpha1.TypeObject: true,
+			v1alpha1.TypeArray:  true,
+			v1alpha1.TypeNull:   true,
+		}
+		if !validTypes[value.Type] {
+			// Invalid type value, skip validation
+			return
+		}
+
+		// Validate the value based on its type
+		// Note: The Value struct fields are marked with omitempty, so they can be
+		// omitted from JSON even when the type is specified. This is valid behavior.
+		switch value.Type {
+		case v1alpha1.TypeString:
+			// String type can have nil String field (omitted from JSON)
+			// This is valid when the JSON only specifies the type
+		case v1alpha1.TypeNumber:
+			// Number type can have nil Number field (omitted from JSON)
+			// This is valid when the JSON only specifies the type
+		case v1alpha1.TypeBool:
+			// Bool type can have nil Bool field (omitted from JSON)
+			// This is valid when the JSON only specifies the type
+		case v1alpha1.TypeObject:
+			// Object type can have nil Object field (omitted from JSON)
+			// This is valid when the JSON only specifies the type
+		case v1alpha1.TypeArray:
+			// Array type can have nil Array field (omitted from JSON)
+			// This is valid when the JSON only specifies the type
+		case v1alpha1.TypeNull:
+			// Null type doesn't need any fields
+		}
```

Comment:
This `switch` statement does nothing in every case. The whole thing is a no-op, and as such it should be removed.